deploy/kubeturbo-operator/helm-charts/templates/configmap.yaml

```diff
@@ -14,12 +14,16 @@ data:
                 "opsManagerUserName": "{{ .Values.restAPIConfig.opsManagerUserName }}",
                 "opsManagerPassword": "{{ .Values.restAPIConfig.opsManagerPassword }}"
             }
+        },
+        "masterNodeDetectors": {
+           "nodeNamePatterns": [ "{{ .Values.masterNodeDetectors.nodeNamePatterns }}" ],
+           "nodeLabels": [ {"key": "{{ .Values.masterNodeDetectors.nodeLabelsKey }}", "value": "{{ .Values.masterNodeDetectors.nodeLabelsValue }}"} ]
+        },
```

Comment:
Eva, this is  good catch, but I think you need to move the extra comma inside the next if statement, because if the next statement is optional, it will result in an invalid yaml again

Reply:
Endre, a summary of our email exchange.  Defaulting to nothing means kubeturbo gets an IP address identifier, which is not useful and user will have to redeploy a cm change, restart pods, wait for grouping, and delete old targets in Turbo UI.  The question is without a targetName, what happens to target identifier should kubeturbo restart - would it create new targets? What does kubeturbo tell the Turbo Server about itself that is consistently unique even with restarts, and provides the user a meaningful cluster representation, without still having to go configure a targetName value in the end?

Reply:
Made the following changes to helm chart in both helm and operator directories:
1. made targetConfig optional, and simplified the if statement
2. handled daemonPod Detectors, as optional, along with masterNodeDetectors, that while not optional, having a default of some value (see values.yaml) is ignored if there are no masters so nothing breaks.
3. updated documentation

Tested successfully:
1. No targetConfig, and with daemonPodDetectors – check
2. No daemonPodDetectors, and with targetConfig – check
3. specified both targetConfig and daemonPodDetectors – check
4. neither targetConfig or daemonPodDetectors - check